src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -59,29 +59,23 @@ public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposab
                 // TODO: We could support reconnecting, like old SignalR does.
                 throw new InvalidOperationException("Connection to redis failed.");
             }
+            _redisServerConnection.PreserveAsyncOrder = true;
             _bus = _redisServerConnection.GetSubscriber();
 
-            var previousBroadcastTask = Task.CompletedTask;
-
             var channelName = typeof(THub).FullName;
             _logger.LogInformation("Subscribing to channel: {channel}", channelName);
             _bus.Subscribe(channelName, async (c, data) =>
             {
-                await previousBroadcastTask;
-
                 _logger.LogTrace("Received message from redis channel {channel}", channelName);
 
                 var message = DeserializeMessage(data);
 
                 // TODO: This isn't going to work when we allow JsonSerializer customization or add Protobuf
-                var tasks = new List<Task>(_connections.Count);
 
                 foreach (var connection in _connections)
                 {
-                    tasks.Add(WriteAsync(connection, message));
+                    await WriteAsync(connection, message);
```

Comment:
Does subscribe support an async `Task` callback? If no, things can still happen in the wrong order since the callback would be async void and the library would think the call is done after the first await.